Formula/postgresql@12.rb

```diff
@@ -81,22 +81,87 @@ def post_install
     return if ENV["CI"]
 
     (var/"log").mkpath
-    (var/"postgres").mkpath
-    unless File.exist? "#{var}/postgres/PG_VERSION"
-      system "#{bin}/initdb", "--locale=C", "-E", "UTF-8", "#{var}/postgres"
+    (var/name).mkpath
+    unless File.exist? "#{var}/#{name}/PG_VERSION"
+      system "#{bin}/initdb", "--locale=C", "-E", "UTF-8", "#{var}/#{name}"
     end
   end
 
+  # Previous versions of this formula used the same data dir as the regular
+  # postgresql formula. So we check whether the versioned data dir exists
+  # and has a PG_VERSION file, which should indicate that the versioned
+  # data dir is in use. Otherwise, returns the old data dir path.
+  def postgresql_datadir
+    if versioned_data_dir_exists?
+      "#{var}/#{name}"
+    else
+      "#{var}/postgres"
```

Comment:
This is repeated in a few places, might be nice extracting it.

Reply:
Adressed in dce8c390c3